src/main/kotlin/no/nav/dagpenger/innsending/løsere/RapporteringPdfBehovLøser.kt

```diff
@@ -0,0 +1,112 @@
+package no.nav.dagpenger.innsending.løsere
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.JsonNodeType
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.slf4j.MDCContext
+import kotlinx.html.BODY
+import kotlinx.html.body
+import kotlinx.html.div
+import kotlinx.html.head
+import kotlinx.html.html
+import kotlinx.html.lang
+import kotlinx.html.stream.createHTML
+import kotlinx.html.title
+import mu.KotlinLogging
+import mu.withLoggingContext
+import no.nav.dagpenger.innsending.ArkiverbartDokument
+import no.nav.dagpenger.innsending.LagretDokument.Companion.behovSvar
+import no.nav.dagpenger.innsending.html.søknadPdfStyle
+import no.nav.dagpenger.innsending.pdf.PdfBuilder
+import no.nav.dagpenger.innsending.pdf.PdfLagring
+import no.nav.dagpenger.innsending.serder.ident
+import no.nav.helse.rapids_rivers.JsonMessage
+import no.nav.helse.rapids_rivers.MessageContext
+import no.nav.helse.rapids_rivers.RapidsConnection
+import no.nav.helse.rapids_rivers.River
+import java.util.UUID
+
+internal class RapporteringPdfBehovLøser(
+    rapidsConnection: RapidsConnection,
+    private val pdfLagring: PdfLagring,
+) : River.PacketListener {
+    companion object {
+        private val logg = KotlinLogging.logger {}
+        private val sikkerlogg = KotlinLogging.logger("tjenestekall")
+        const val BEHOV = "OpprettPdfForRapportering"
+    }
+
+    init {
+        River(rapidsConnection).apply {
+            validate { it.demandValue("@event_name", "behov") }
+            validate { it.demandAll("@behov", listOf(BEHOV)) }
+            validate { it.rejectKey("@løsning") }
+            validate { it.requireKey("ident", "periodeId", "journalpostId", "json") }
+        }.register(this)
+    }
+
+    override fun onPacket(packet: JsonMessage, context: MessageContext) {
+        val ident = packet.ident()
+        val periodeId = packet["periodeId"].asText().let { UUID.fromString(it) }
+        val journalpostId = packet["journalpostId"].asText()
+        val json = jacksonObjectMapper().readTree(packet["json"].asText())
+
+        withLoggingContext(
+            "periodeId" to periodeId.toString(),
+            "journalpostId" to journalpostId,
+        ) {
+            try {
+                runBlocking(MDCContext()) {
+                    logg.info("Mottok behov for PDF av rapportering")
+
+                    val html = createHTML(prettyPrint = false, xhtmlCompatible = false).html {
```

Comment:
Jeg synes at laging av HTML og PDF burde trekkes ut i egen klasse/funksjon og med test.
Om ikke annet bare for å sjekke at det faktisk er gyldig html og pdf som lages. 
Samt at man enkelt kan ha manuell test for å se om pdf ser bra ut.